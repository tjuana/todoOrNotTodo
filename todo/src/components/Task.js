import React from "react";
import TaskInput from "./TaskInput";

const Task = ({ edit, task, ...props }) => {
  const ActionBtn = () => (
    <div className="action-btn">
      {task.done ? (
        <span aria-label="done" role="img" onClick={props.deleteTask}>
          âœ…
        </span>
      ) : (
        <span aria-label="delete" role="img" onClick={props.doneTask}>
          ðŸŽ¯
        </span>
      )}
    </div>
  );

  const className = "task " + (task.done ? "task-done" : "");
  var viewStyle = {};
  var editStyle = {};

  if (edit) {
    viewStyle.display = "none";
  } else {
    editStyle.display = "none";
  }

  return (
    <div className={className} onDoubleClick={props.editTask}>
      <p style={viewStyle} >{task.title}</p>
      <input
        style={editStyle}
        className="task-input"
        type="text"
        value={TaskInput.input}
      ></input>
      <ActionBtn></ActionBtn>
    </div>
  );
};

export default Task;
