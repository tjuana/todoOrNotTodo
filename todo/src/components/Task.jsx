import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './Task.module.css';
import {
  doneTask, deleteTask, editTaskView, editTaskInput,
} from '../redux/todoActions.jsx';

const Task = ({
  editing, id, isDone, title, del, doneTaskView, editView, editInput,
}) => {
  const handleClick = () => {
    if (isDone) {
      del(id);
    } else {
      doneTaskView(id);
    }
  };

  const ActionBtn = () => (
    <button className={styles.actionbtn} onClick={handleClick} type="submit">
      {!isDone ? (
        <span aria-label="delete" role="img">
          âœ…
        </span>
      ) : (
        <span aria-label="done" role="img">
          ðŸŽ¯
        </span>
      )}
    </button>
  );

  let inputValue = title;
  const viewStyle = {};
  const editStyle = {};

  if (editing) {
    viewStyle.display = 'none';
  } else {
    editStyle.display = 'none';
  }

  const handleChange = (event) => {
    inputValue = event.target.value;
    editInput(id, inputValue);
  };

  const handleEditView = () => {
    editView(id);
  };

  return (
    <div className={!isDone ? styles.task : styles.taskdone} onDoubleClick={handleEditView}>
      <p style={viewStyle}>{title}</p>
      <input
        onChange={handleChange}
        style={editStyle}
        className={styles.input}
        type="text"
        value={inputValue}
      />
      <ActionBtn />
    </div>
  );
};

Task.propTypes = {
  doneTaskView: PropTypes.func.isRequired,
  del: PropTypes.func.isRequired,
  editing: PropTypes.bool.isRequired,
  editView: PropTypes.func.isRequired,
  editInput: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  isDone: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
};

const mapDispatchToProps = {
  del: deleteTask,
  doneTaskView: doneTask,
  editView: editTaskView,
  editInput: editTaskInput,
};

export default connect(null, mapDispatchToProps)(Task);
