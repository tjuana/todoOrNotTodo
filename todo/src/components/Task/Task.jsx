import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './Task.module.css';
import { editTaskView, editTaskInput } from '../../action/todoActions.jsx';
import ActionBtn from '../ActionBtn/ActionBtn.jsx';

const Task = ({
  isEditMode, id, isDone, title, editView, editInput,
}) => {
  const [inputVal, setInputVal] = useState(title);

  useEffect(() => {
    editInput(id, inputVal);
  }, [inputVal, id, editInput]);

  const handleChange = (event) => {
    event.preventDefault();
    setInputVal(event.target.value);
  };

  const handleEditView = () => {
    editView(id);
  };

  return (
    <div className={cx(styles.task, { [styles.taskdone]: isDone })}>

      <ul>
        <li onDoubleClick={handleEditView}>
          <p className={cx({ [styles.hidden]: isEditMode })}>{title}</p>
          <form onSubmit={handleEditView}>
            <input
              onChange={handleChange}
              className={cx(styles.input, { [styles.hidden]: !isEditMode })}
              type="text"
              value={inputVal}
            />
          </form>
        </li>
      </ul>
      <ActionBtn
        isDone={isDone}
        id={id}
      />
    </div>

  );
};

Task.propTypes = {
  isEditMode: PropTypes.bool.isRequired,
  editView: PropTypes.func.isRequired,
  editInput: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  isDone: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
};

const mapDispatchToProps = {
  editView: editTaskView,
  editInput: editTaskInput,
};

export default connect(null, mapDispatchToProps)(Task);

export { Task as TaskList };
