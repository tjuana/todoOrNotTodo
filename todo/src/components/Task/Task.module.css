.task {
  display: flex;
  padding: 0px;
  margin: 5px;
  border-radius: 5px;
  justify-content: space-between;
}

.task:hover {
  background-color: #04395e3d;
}

.task p {
  font-size: 19px;
  font-weight: 750;
  margin: 0;
  color: #0d568a;
}

.input {
  font-size: 20px;
  font-weight: 750;
  margin: 0;
  color:#0d568a;
  background-color: aliceblue;
  text-align: center;
  border-radius: 15px;
}

.taskdone {
  background-color: #00000027;
}

.taskdone p {
  color: #00000050;
}

.taskdone p {
  font-size: 20px;
  font-weight: 750;
  margin: 0;
  color: #0d568a;
}

.taskdone {
  display: flex;
  padding: 0px;
  margin: 5px;
  border-radius: 5px;
  justify-content: space-between;
}

.hidden {
  display: none;
}
