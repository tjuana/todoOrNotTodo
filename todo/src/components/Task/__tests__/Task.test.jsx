import { shallow, mount } from 'enzyme';
import React from 'react';
import { TaskList } from '../Task.jsx';

describe('Task lits component test', () => {
  let props;
  let wrap;
  const event = { preventDefault: () => {} };

  beforeEach(() => {
    props = {
      editInput: jest.fn(),
      editView: jest.fn(),
      id: 0,
      isEditMode: false,
      isDone: false,
      title: 'blabla',
    };
    wrap = shallow(<TaskList {...props} />);
    jest.spyOn(event, 'preventDefault');
  });

  it('should render init state', () => {
    expect(wrap.find('div').length).toBe(1);
    expect(wrap.find('p').length).toBe(1);
    expect(wrap.find('input').length).toBe(1);
  });

  it('should handleDoubleClick and text of p', () => {
    wrap.find('li').simulate('DoubleClick');
    expect(props.editView).toHaveBeenCalledTimes(1);
    expect(wrap.find('p').text()).toBe('blabla');
    wrap.find('input').simulate('change', { target: { value: 'change' } });
    expect(wrap.find('input').prop('value')).toBe('change');
  });

  it('should check isDone', () => {
    expect(wrap.find('div').hasClass('task')).toBe(true);
  });

  it('should check isEditMode', () => {
    const wrapper = shallow(<TaskList {...props} isEditMode />);
    expect(wrapper.find('p').hasClass('hidden')).toBe(true);
    expect(wrapper.find('input').hasClass('input')).toBe(true);
  });

  it('should check isDone', () => {
    const wrapper = shallow(<TaskList {...props} isDone />);
    expect(wrapper.find('div').hasClass('taskdone')).toBe(true);
  });
});
