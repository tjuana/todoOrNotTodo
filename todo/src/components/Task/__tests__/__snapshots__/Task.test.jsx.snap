// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task lits component test should Task snap 1`] = `
<div
  className="task"
>
  <ul>
    <li
      onDoubleClick={[Function]}
    >
      <p
        className={false}
      >
        blabla2
      </p>
      <form
        onSubmit={[Function]}
      >
        <input
          className="hidden"
          onChange={[Function]}
          type="text"
          value="blabla2"
        />
      </form>
    </li>
  </ul>
  <button
    className="actionbtn"
    onClick={[Function]}
    type="submit"
  >
    <span
      aria-label="delete"
      role="img"
    >
      âœ…
    </span>
  </button>
</div>
`;
